sensors/nxt_i2c_sensor_core.c

```diff
@@ -377,6 +399,18 @@ static int nxt_i2c_sensor_detect(struct i2c_client *client,
 #endif
 
 	/*
+	 * NXT Temperature sensor detection is handled here.
+	 */
+	if (client->addr == 0x4c) {
+		ret = nxt_i2c_sensor_detect_lego_temp(client);
+		if (ret)
```

Comment:
You can also get an error from the i2c function if we fail to communicate. The first option might be better though.